networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java

```diff
@@ -112,10 +120,27 @@ public void onIncomingMessage(
                               if (validationResult.isPresent()) {
                                 return SafeFuture.failedFuture(validationResult.get());
                               }
-                              return block.map(callback::respond).orElse(SafeFuture.COMPLETE);
+                              return block
+                                  .map(
+                                      signedBeaconBlock ->
+                                          callback
+                                              .respond(signedBeaconBlock)
+                                              .thenRun(sentBlocks::incrementAndGet))
+                                  .orElse(SafeFuture.COMPLETE);
                             }));
       }
-      future.finish(callback::completeSuccessfully, err -> handleError(callback, err));
+      future.finish(
+          () -> {
+            if (sentBlocks.get() != message.size()) {
+              peer.adjustBlockRequests(blockRequestsApproval.get(), sentBlocks.get());
+            }
+            totalBlocksRequestedCounter.inc(sentBlocks.get());
```

Comment:
You forgot to move counter here outside to requested, not sent values

Reply:
Done